geotrek/tourism/parsers.py

```diff
@@ -98,34 +89,6 @@ class ApidaeParser(AttachmentParserMixin, Parser):
         'illustrations'
     ]
 
-    @property
-    def items(self):
-        if self.nb == 0:
-            return []
-        return self.root['objetsTouristiques']
-
-    def next_row(self):
-        while True:
-            params = {
-                'apiKey': self.api_key,
-                'projetId': self.project_id,
-                'selectionIds': [self.selection_id],
-                'count': self.size,
-                'first': self.skip,
-                'responseFields': self.responseFields
-            }
-            response = self.request_or_retry(self.url, params={'query': json.dumps(params)})
-            self.root = response.json()
-            self.nb = int(self.root['numFound'])
-            for row in self.items:
-                yield row
-            self.skip += self.size
-            if self.skip >= self.nb:
-                return
-
-    def normalize_field_name(self, name):
-        return name
-
     def filter_eid(self, src, val):
```

Comment:
I tried to move only the behavior needed by the `ApidaeTrekParser` which is the way to build the URL and to request the API. I could move the whole class but `ApidaeTrekParser` would inherit attributes or filter hooks it does not need.
What troubles you in the fact of splitting this class?

Reply:
OK, good point for the forgotten `filter_eid`. I will not place it in the base class as it is related to field mapping which is the concern of subclasses here. But I have added it to `geotrek.trekking.parsers.ApidaeTrekParser` just in case.